server/initializers/sequelize.js

```diff
@@ -187,33 +218,57 @@ module.exports = upgradeInitializer('ah17', {
 
   startPriority: 101, // aligned with actionhero's redis initializer
   start: function (api, next) {
-    api.sequelize.connect(function (err) {
-      if (err) {
-        return next(err)
-      }
-
-      api.sequelize.autoMigrate(function (err) {
-        if (err) {
-          console.error(err)
-          return next(err)
-        }
-        api.sequelize.loadFixtures(next)
+    Promise.resolve()
+      .then(() => {
+        if (!api.config.sequelize.testing) return
+        api.log('Creating database...', 'debug')
+        return api.sequelize.createDb()
+          .then(() => api.log('Creating database... done', 'debug'))
+      })
+      .then(() => new Promise((resolve, reject) => api.sequelize.connect((err) => err ? reject(err) : resolve())))
+      .then(() => {
+        if (!api.config.sequelize.testing) return
+        api.log('Starting global transaction...', 'debug')
+        return api.sequelize.sequelize.transaction()
+          .then(function (tx) {
+            api.log('Starting global transaction... done', 'debug')
+            api.sequelize.sequelize.options.query.transaction = tx
+          })
+      })
+      .then(() => new Promise((resolve, reject) => { api.sequelize.autoMigrate((err) => err ? reject(err) : resolve()) }))
+      .then(() => new Promise((resolve, reject) => { api.sequelize.loadFixtures((err) => err ? reject(err) : resolve()) }))
+      .then(() => api.log(`Connected to ${api.config.sequelize.dialect}://${api.config.sequelize.host}:${api.config.sequelize.port}/${api.config.sequelize.database}`, 'info'))
+      .then(() => next(), (err) => {
+        console.error(err)
+        next(err)
       })
-    })
   },
 
   stopPriority: 99999, // aligned with actionhero's redis initializer
   stop: function (api, next) {
-    api.sequelize.sequelize.close()
-      .then(function () {
-        next()
-      }, function (err) {
+    Promise.resolve()
+      .then(() => {
+        if (!api.config.sequelize.testing) return
+        api.log('Rolling back global transaction...', 'debug')
+        return api.sequelize.sequelize.options.query.transaction.rollback()
+          .then(() => {
+            api.log('Rolling back global transaction... done', 'debug')
+          })
+      })
+      .then(() => api.sequelize.sequelize.close())
+      .then(() => {
+        if (!api.config.sequelize.testing) return
+        return api.sequelize.dropDb()
+      })
+      .then(() => next(), (err) => {
+        console.error(err)
         next(err)
       })
   }
 })
 
 function checkMetaOldSchema (api) {
+  if (api.config.sequelize.testing) return Promise.resolve()
```

Comment:
broken query rollback's the transaction, so in testing when a new db is used we need to skip this as the `SequelizeMeta` is not there